### PR TITLE
Bring back lgTemplate function

### PR DESCRIPTION
**Pull request by boydc2014**
In normal case, use templateName(args) is more straightforward, but when you want to call a template without hard-coding the name, lgTemplate function comes in handy.  It works like reflection. 

So bring it back.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs [hunk 1]
```diff
@@ -392,9 +396,61 @@ private EvaluatorLookup CustomizedEvaluatorLookup(EvaluatorLookup baseLookup)
                 return new ExpressionEvaluator(name, BuiltInFunctions.Apply(this.TemplateEvaluator(name)), ReturnType.Object, this.ValidTemplateReference);
             }
 
+            const string lgTempalte = "lgTemplate";
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs]: Spelling

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs [hunk 2]
```diff
@@ -392,9 +396,61 @@ private EvaluatorLookup CustomizedEvaluatorLookup(EvaluatorLookup baseLookup)
...
+
             return baseLookup(name);
         };
 
+        // Evaluator for lgTemplate(templateName, ...args) 
+        // normal case we can just use templateName(...args), but lgTemplate is particularly useful when the template name is not pre-known
+        private Func<IReadOnlyList<object>, object> LgTemplate()
+        => (IReadOnlyList<object> args) =>
+        {
+            var templateName = args[0].ToString();
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs]: Shouldn't this be an expression to evaluate or had that already been done?

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs [hunk 3]
```diff
@@ -392,9 +396,61 @@ private EvaluatorLookup CustomizedEvaluatorLookup(EvaluatorLookup baseLookup)
...
+            }
+
+            // Validate more if the name is string constant
+            if (children0.Type == ExpressionType.Constant)
+            {
+                var templateName = (children0 as Constant).Value.ToString();
+                if (!this.TemplateMap.ContainsKey(templateName))
+                {
+                    throw new Exception($"no such template '{templateName}' to call in {expression}");
+                }
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs]: Isn't the exception standard to start with capital letter?

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs (hunk 2 above)
Comment (boydc2014) [reply libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs]: Builtin.Apply already did that

Comment (boydc2014) [reply libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs]: It's you Builtin.Apply did the work :)

**Pull request merged**